DOC: Updated pointer spacing for consistency.

madphysicist opened the pull request:
I expect that this change will be pretty non-controversial. It was bugging me that the space came before the star in some cases and after in others. Then I went through the tutorial for something I needed, and ended up fixing some warnings and code formatting issued.

Ended up fixing what looks like a copy-and-paste bug in the tutorial code.

With commit e2516f1, this PR depends on #20689

madphysicist committed:
DOC: Updated pointer spacing for consistency.

Mukulikaa commented:
Thanks @madphysicist, you seem to have got them all. I am not sure if these should have spaces as well:
https://github.com/numpy/numpy/blame/main/doc/source/user/c-info.ufunc-tutorial.rst#L875-L877

madphysicist committed:
DOC: Additional modifications to ufunc tutorial

- Couple of extra imports removed
- Couple of warnings about `const` eliminated
- Some whitespace made more conistent
- etc.

madphysicist committed:
DOC: Missed a couple of spaces

madphysicist commented:
@Mukulikaa. I made some additional changes, including the ones you suggest. Thanks for the review. Please let me know if I missed anything.

madphysicist committed:
DOC: Bug in half-float computation example

madphysicist committed:
DOC: Removed unnecessary references to `data`.

This commit makes the PR depend on #20689

mattip commented on doc/source/user/c-info.ufunc-tutorial.rst:
```diff
@@ -242,13 +239,14 @@ Example NumPy ufunc for one dtype
-the ufunc are marked with /\*BEGIN main ufunc computation\*/ and
-/\*END main ufunc computation\*/. The code in between those lines is
+the ufunc are marked with ``/\* BEGIN main ufunc computation \*/`` and
+``/\* END main ufunc computation \*/``. The code in between those lines is
```
Not for this PR, but at some point we should also point out that C++ templating can be used.

mattip commented:
Thanks @madphysicist

madphysicist commented:
@mattip technically, the example code is invalid until #20689 gets merged. The documentation says it's valid, but it will currently segfault everywhere the `data` argument to `PyUFunc_FromFuncAndData` is NULL.
